independent-projects/resteasy-reactive/server/vertx/src/main/java/org/jboss/resteasy/reactive/server/vertx/ResteasyReactiveOutputStream.java

```diff
@@ -87,8 +87,12 @@ public void write(ByteBuf data, boolean last) throws IOException {
                     if (overflow == null) {
                         overflow = new ByteArrayOutputStream();
                     }
-                    overflow.write(data.array(), data.arrayOffset() + data.readerIndex(),
-                            data.arrayOffset() + data.writerIndex());
+                    if (data.hasArray()) {
+                        overflow.write(data.array());
```

Comment:
This double write does not look correct, although I don't think the code path will actually be used at the moment as for now we only allocate direct buffers.

Reply:
Oops! The first write is left over... Let me fix it

Reply:
Fixed